pkg/controllers/core/deployer_controller.go

```diff
@@ -557,6 +558,38 @@ func (r *DeployerReconciler) constructServiceForDeployer(deployer *corev1alpha1.
 	// If a service name was provided, use it, otherwise generate one based on the deployer name.
 	if deployer.Spec.ServiceName != "" {
 		service.ObjectMeta.Name = deployer.Spec.ServiceName
+
+		// Detect service name collision.
+		collidingService, err := r.getService(deployer.Namespace, deployer.Spec.ServiceName)
+		if err != nil {
+			return nil, err
+		}
+		if collidingService != nil {
+			// Diagnose service name collisions. If the colliding service was created by a deployer, include that in the
```

Comment:
it doesn't matter why the resource collided. The only thing that matters is the resource already exists and the deployer being reconciled doesn't own it.

Reply:
Ok, so capturing the fact of a collision should give the user sufficient information to fix the problem.